modules/apps/portal-scheduler/portal-scheduler-multiple/src/main/java/com/liferay/portal/scheduler/multiple/internal/ClusterableProxyFactory.java

```diff
@@ -56,7 +55,7 @@ public Object invoke(Object proxy, Method method, Object[] arguments)
 			Clusterable clusterable = _getClusterable(
 				method, _targetObject.getClass());
 
-			if (clusterable == NullClusterable.NULL_CLUSTERABLE) {
+			if (clusterable == null) {
```

Comment:
@dantewang @yuhai 
NullClusterable.NULL_CLUSTERABLE here is used as a placeholder for null in cache. 

After your change, the cache only stores the class with clusterable annotation, for those not having clusterable annotation, you will have to check it again and again. This will make the cache useless cause most of classes do not have annotation, and the cache is not storing the annotation result for them, thus most of checking of cache will miss.

In this case, you need to inline NullClusterable into this class.


Tina.